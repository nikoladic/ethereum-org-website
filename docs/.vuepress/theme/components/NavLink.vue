// Do not use classes in this file // Pass classes into this via NavLinks.vue
<template>
  <router-link
    ref="el"
    :to="link"
    :class="childClass"
    v-if="!isExternal(link)"
    :exact="exact"
    @click.native="$emit('nav-toggle', false)"
    >{{ item.text }}</router-link
  >
  <a
    v-else-if="!item.hideMobile"
    :href="link"
    :class="childClass + ' nav-link hide-icon'"
    v-on:click.native="$emit('nav-toggle', false)"
    :target="isMailto(link) || isTel(link) ? null : '_blank'"
    :rel="isMailto(link) || isTel(link) ? null : 'noopener noreferrer'"
  >
    {{ item.text }}
    <OutboundLink />
  </a>
</template>

<script>
import { isExternal, isMailto, isTel, ensureExt } from '../utils/util'

export default {
  props: {
    item: {
      required: true
    },
    childClass: {
      type: String,
      required: false
    },
    closeMenu: {
      type: Function
    }
  },
  computed: {
    link() {
      return ensureExt(this.item.link)
    },
    exact() {
      if (this.$site.locales) {
        return Object.keys(this.$site.locales).some(rootLink => {
          return rootLink === this.link
        })
      }
      return this.link === '/'
    }
  },

  methods: {
    isExternal,
    isMailto,
    isTel
  }
}
</script>
