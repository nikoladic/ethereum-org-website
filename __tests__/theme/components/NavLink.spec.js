import { shallowMount, mount } from '@vue/test-utils'
import NavLink from "@/theme/components/NavLink.vue"


describe('As user i can see a link', () => {
  test('rendered as normal link', () => {
    const wrapper = mount(NavLink, {
      propsData: {item: { text: "ethereum", link: "https://etherum.org" }}
    });

    expect(wrapper.element).toMatchSnapshot()
  });

  test('rendered as external link', () => {
    const wrapper = shallowMount(NavLink, {
      propsData: {
        item: {
          text: "Wallet",
          link: "wallet://address.com/?account=199283"
        }
      },
      mocks: {
        $site: {
          locales: {
            "es": {label: 'EspaÃ±ol', lang: 'es'},
            "en": {label: 'English', lang: 'en'}
          }
        }
      }
    });

    expect(wrapper.element).toMatchSnapshot()
  });

  test('rendered as telephone link', () => {
    const wrapper = mount(NavLink, {
      propsData: {item: { text: "phone", link: "tel:+5215567233201" }}
    });

    expect(wrapper.element).toMatchSnapshot()
  });

  test('rendered as email link', () => {
    const wrapper = mount(NavLink, {
      propsData: { item: {text: "email", link: "mailto:contact@ethereum.org"}}
    });

    expect(wrapper.element).toMatchSnapshot()
  });
});