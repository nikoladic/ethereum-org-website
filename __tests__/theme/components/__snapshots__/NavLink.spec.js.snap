// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`As user i can see a link rendered as email link 1`] = `
<a
  class="nav-link external"
  href="mailto:contact@ethereum.org"
>
  
  email
  
  <outboundlink />
</a>
`;

exports[`As user i can see a link rendered as external link 1`] = `
<router-link
  class="nav-link"
  to="wallet://address.com/?account=199283.html"
>
  Wallet
</router-link>
`;

exports[`As user i can see a link rendered as normal link 1`] = `
<a
  class="nav-link external"
  href="https://etherum.org"
  rel="noopener noreferrer"
  target="_blank"
>
  
  ethereum
  
  <outboundlink />
</a>
`;

exports[`As user i can see a link rendered as telephone link 1`] = `
<a
  class="nav-link external"
  href="tel:+5215567233201"
>
  
  phone
  
  <outboundlink />
</a>
`;
